src/Refitter.Core/ParameterExtractor.cs

```diff
@@ -142,4 +148,84 @@ private static string JoinAttributes(params string[] attributes)
 
     private static string FindSupportedType(string typeName) =>
         typeName == "FileResponse" ? "StreamPart" : typeName;
+
+    private static List<string> GetQueryParameters(CSharpOperationModel operationModel, RefitGeneratorSettings settings, string dynamicQuerystringParameterType, out string? dynamicQuerystringParameters)
+    {
+        List<string>? parameters = null;
+        var dynamicQuerystringParametersCodeBuilder = new StringBuilder();
+
+        if (settings.DynamicQuerystringParametersThreshold >= 2)
+        {
+            var operationParameters = operationModel.Parameters
+                .Where(p => p.Kind == OpenApiParameterKind.Query)
+                .ToList();
+
+            if (operationParameters.Count >= settings.DynamicQuerystringParametersThreshold)
+            {
+                var modifier = settings.TypeAccessibility.ToString().ToLowerInvariant();
+                var isRecord = settings.ImmutableRecords ||
+                               settings.CodeGeneratorSettings?.GenerateNativeRecords is true;
+                var classStyle = isRecord
+                    ? "record"
+                    : "class";
+                var setterStyle = isRecord
+                    ? "init"
+                    : "set";
+
+                var propertiesCodeBuilder = new StringBuilder();
+                var allNullable = true;
+                foreach (var operationParameter in operationParameters)
+                {
+                    var attributes = $"{JoinAttributes(GetQueryAttribute(operationParameter, settings), GetAliasAsAttribute(operationParameter))}";
+                    var propertyType = GetQueryParameterType(operationParameter, settings);
+                    allNullable = allNullable && propertyType.EndsWith("?");
```

Comment:
I pulled the branch locally and tried it out. I get some build warnings on the generated code

```
CS8669: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source
```

The `#nullable enable` directive is added to the generated code in the [RefitGenerator](https://github.com/christianhelle/refitter/blob/be76c53f63694c5ddb67ef136b6256fe18996bde/src/Refitter.Core/RefitGenerator.cs#L180) class, if the `OptionalParameters` option is enabled

Reply:
The `#nullable enable` is also added from [ApizrRegistrationGenerator](https://github.com/christianhelle/refitter/blob/be76c53f63694c5ddb67ef136b6256fe18996bde/src/Refitter.Core/ApizrRegistrationGenerator.cs#L389) and [DependencyInjectionGenerator](https://github.com/christianhelle/refitter/blob/be76c53f63694c5ddb67ef136b6256fe18996bde/src/Refitter.Core/DependencyInjectionGenerator.cs#L63)